File: opencv/Task1_basic_operates/Code/phone.py
import cv2

# 读取图片
img = cv2.imread('all.jpg')

# 显示图片，让用户手动框选区域
r = cv2.selectROI("Image", img)

# 提取选中的区域（手机部分）
cropped = img[int(r[1]):int(r[1] + r[3]), int(r[0]):int(r[0] + r[2])]

# 保存截取后的图片为phone.jpg
cv2.imwrite('phone.jpg', cropped)

# 关闭显示窗口
cv2.destroyAllWindows()

File: Task 4 OpenCV Cpp_黄世昌/Code/phone.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main() {

    cv::Mat image = cv::imread("D:\\»ÆÊÀ²ý\\Desktop\\Task_4\\Task_4\\all.jpg");


    if (image.empty()) {
        std::cerr << "Error: Could not load image." << std::endl;
        return -1;
    }

    cv::Rect roi(50, 100, 500, 300);
    cv::Mat phone = image(roi);

    cv::imwrite("phone.jpg", phone);
    std::cout << "Phone image saved as phone.jpg" << std::endl;

    return 0;
}